cli/src/hooks/init/compatibility.ts

```diff
@@ -0,0 +1,94 @@
+import { Hook } from '@oclif/core';
+import { readFile, stat, writeFile } from 'fs/promises';
+import semver from 'semver';
+import { mkdir } from 'fs';
+import path from 'path';
+import fetch from 'node-fetch';
+
+export type Packages = 'cli' | 'sdk';
+export type Compatibility = {
+  [key in Packages]: {
+    latest: string;
+    compatibility: { range: string; compatible: boolean; error?: string }[];
+  };
+};
+export type Latest = { latest: { [key in Packages]: string } };
+export type PackageJson = { dependencies: Record<string, string> };
+
+const hook: Hook<'init'> = async function (_options) {
+  const ONE_DAY = 1000 * 60 * 60 * 24 * 1;
+  const dir = path.join(process.cwd(), '.xata', 'version');
+  const latestFile = `${dir}/version.json`;
+  const compatibilityFile = `${dir}/compatibility.json`;
+  const compatibilityUri = 'https://raw.githubusercontent.com/xataio/client-ts/main/compatibility.json';
+
+  const displayWarning = async () => {
+    await check('cli', this.config.version);
+    const sdkVersion = await getSdkVersion();
+    sdkVersion && (await check('sdk', sdkVersion));
+  };
+
+  const check = async (pkg: 'cli' | 'sdk', currentVersion: string) => {
+    // Latest version check
+    const tag: Latest = JSON.parse(await readFile(latestFile, 'utf-8'));
+    const updateAvailable = semver.lt(this.config.version, tag.latest[pkg]);
+    if (updateAvailable)
+      this.log(
+        `✨ A newer version of the Xata ${pkg.toUpperCase()} is now available: ${
+          tag.latest[pkg]
+        }. You are currently using version: ${currentVersion}`
+      );
+    // Compatibility check
+    const compatibility: Compatibility = JSON.parse(await readFile(compatibilityFile, 'utf-8'));
+    const compatibleRange = compatibility[pkg].compatibility
+      .filter((v) => v.compatible)
+      .map((v) => v.range)
+      .join('||');
```

Comment:
There doesn't seem to be a utility for getting rid of this join, even a Range object expects a || separated string